modules/dhcp_common/subnet.rb

```diff
@@ -62,28 +64,19 @@ def netmask_to_i
       @netmask_to_i ||= Proxy::DHCP.ipv4_to_i(netmask)
     end
 
-    def get_index_and_lock filename
+    def get_index filename
       # Store for use in the unlock method
       @filename = "#{Dir::tmpdir}/#{filename}"
-      @lockfile = "#{@filename}.lock"
-
-      # Loop if the file is locked
-      Timeout::timeout(30) { sleep 0.1 while File.exist? @lockfile }
-
-      # Touch the lock the file
-      File.open(@lockfile, "w") {}
-
       @file = File.new(@filename,'r+') rescue File.new(@filename,'w+')
-
       # this returns the index in the file
       return @file.readlines.first.to_i rescue 0
     end
 
-    def write_index_and_unlock index
+    def write_index index
       @file.reopen(@filename,'w')
       @file.write index
+    ensure
       @file.close
-      File.delete @lockfile
     end
```

Comment:
One change needed tho, since we do not use the file for locking now, there is no point of having both file and filename as member variables, I'd simply rewrite read and write methods using safe block file operations.